OrbitQt/main.cpp

```diff
@@ -208,6 +217,101 @@ static outcome::result<void> RunUiInstance(
+  CHECK(deployment_configuration.has_value());
+
+  const GrpcPort grpc_port{/*.grpc_port =*/absl::GetFlag(FLAGS_grpc_port)};
+
+  std::unique_ptr<MainThreadExecutor> main_thread_executor = CreateMainThreadExecutor();
+  orbit_qt::SshConnectionArtifacts ssh_connection_artifacts{ssh_context, grpc_port,
+                                                            &(deployment_configuration.value())};
+
+  std::optional<orbit_qt::ConnectionConfiguration> connection_config;
+
+  while (true) {
+    {
+      orbit_qt::ProfilingTargetDialog target_dialog{
+          &ssh_connection_artifacts, main_thread_executor.get(), std::move(connection_config)};
+      connection_config = target_dialog.Exec();
+
+      if (!connection_config.has_value()) {
+        // User closed dialog
+        break;
+      }
+    }
+
+    GOrbitApp = OrbitApp::Create(main_thread_executor.get());
+
+    int application_return_code = 0;
+
+    {  // Scoping of QT UI Resources
+
+      ServiceDeployManager* service_deploy_manager_ptr = nullptr;
+      std::visit(
+          [&service_deploy_manager_ptr](auto&& target) {
+            using Target = std::decay_t<decltype(target)>;
+            if constexpr (std::is_same_v<Target, orbit_qt::StadiaProfilingTarget>) {
+              service_deploy_manager_ptr = target.GetConnection()->GetServiceDeployManager();
+            } else if constexpr (std::is_same_v<Target, orbit_qt::LocalTarget>) {
+            } else if constexpr (std::is_same_v<Target, orbit_qt::FileTarget>) {
+            } else {
+              UNREACHABLE();
+            }
+          },
+          connection_config.value());
+
+      constexpr uint32_t kDefaultFontSize = 14;
+
+      OrbitMainWindow w(app, std::move(connection_config.value()), kDefaultFontSize);
+
+      // "resize" is required to make "showMaximized" work properly.
+      w.resize(1280, 720);
+      w.showMaximized();
+
+      QMessageBox box(QMessageBox::Critical, QApplication::applicationName(), "", QMessageBox::Ok);
+      auto error_handler = [&]() -> ScopedConnection {
+        if (service_deploy_manager_ptr == nullptr) {
+          return ScopedConnection();
+        }
+
+        return OrbitSshQt::ScopedConnection{QObject::connect(
+            service_deploy_manager_ptr, &ServiceDeployManager::socketErrorOccurred,
+            service_deploy_manager_ptr, [&](std::error_code error) {
+              main_thread_executor->Schedule([&box, &w, error] {
+                box.setText(
+                    QString("Connection error: %1").arg(QString::fromStdString(error.message())));
+                box.exec();
+                w.close();
+                QApplication::exit(1);
+              });
+            })};
+      }();
+
+      application_return_code = QApplication::exec();
+
+      connection_config = w.ClearConnectionConfiguration();
+    }
+
+    Orbit_ImGui_Shutdown();
+    GOrbitApp->OnExit();
+
+    if (application_return_code == 0) {
```

Comment:
Could we add constants here or doesn't have any sense?

Reply:
done